### DISCUSSION TITLE
[custom-plugin-lib] Graceful shutdown - fastify.close is not a function

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#326] [Q&A] [answered]
### Which section?

Marketplace

### Which is the version?

v11.6.2

### Description

Hi guys,
we have a Node.JS microservice that uses the Mia-Platform [_custom-plugin-lib_](https://github.com/mia-platform/custom-plugin-lib) with Fastify.

We noticed a weird behaviour when the service was shut down by Kubernetes because of multiple `livenessProbe` failures.
When multiple `livenessProbe` fail, Kubernetes send a `SIGTERM` to the pod that performs a graceful shutdown, calling the `fastify.close()` function. Despite the `.close` function exists, the last log of the service is the following one:
```
{"level":50,"time":1696341835306,"pid":7,"hostname":"redirect-manager-64b966f9df-96598","debugInfo":{"status":"KO","name":"redirect-manager","version":"3.3.0-rc.1"},"msg":"Service status is not ok"}
/home/node/app/node_modules/@mia-platform/lc39/lib/launch-fastify.js:172
      fastify.close()
              ^

TypeError: fastify.close is not a function
    at Timeout._onTimeout (/home/node/app/node_modules/@mia-platform/lc39/lib/launch-fastify.js:172:15)
    at listOnTimeout (node:internal/timers:569:17)
    at process.processTimers (node:internal/timers:512:7)
```

We are using Node.JS `v18.18.0` and the version v6.0.0 of the `custom-plugin-lib`.

Did you ever have this error?

Thanks.

**Comment by fredmaggiowski** [comment, marked as answer]
Hey, while I don't have an answer to this right now (I have to investigate it a little bit) I'd create a new issue on the [custom plugin lib](https://github.com/mia-platform/custom-plugin-lib) and maybe link the discussion (or close it)

EDIT: maybe this issue is more relevant to [lc39](https://github.com/mia-platform/lc39)

**Reply by frengooo** [reply]
Thanks @fredmaggiowski, I'll open it on `lc39`.

**Reply by fredmaggiowski** [reply]
For completion here the issue link you opened: https://github.com/mia-platform/lc39/issues/308

**Comment by davidebianchi** [comment]
Yes, I already saw it. It should be opened as a bug inside the lc39 repo, and find a resolution for the bug.

**Reply by frengooo** [reply]
Thanks @davidebianchi, I'll open it on `lc39`.
